mesa/visualization/ModularVisualization.py

```diff
@@ -223,19 +223,24 @@ def on_message(self, message):
         if self.application.verbose:
             print(message)
         msg = tornado.escape.json_decode(message)
+        try:
+            msg_type = msg["type"]
+        except KeyError:
+            print("Unexpected message. Key 'type' missing.")
```

Comment:
I haven't encountered this branch point yet.
I think the safeguard is not necessary. Printing this error message instead of raising the exception results in an information loss about the location in which the error happens.
Replacing with `msg_type = msg["type"]` should do.

Reply:
> I haven't encountered this branch point yet.

I ran into this, as I was trying out various changes in `mesa/visualization/templates/js/runcontrol.js`.
When I sent a message that didn't contain a key "type" 

=>

* The SocketHandler threw an error
* The Tornado server shut down.

If we add an error handling, the server will continue running despite of the unexpected message.
I'm not sure whether this is a good or bad thing.

> Printing this error message instead of raising the exception results in an information loss about the location in which the error happens.

That's a good point.
Is there more info that should be included in the error message?
Or do you think that raising the exception and shutting down the server is the way to go?

Reply:
> When I sent a message that didn't contain a key "type"

This is a change that is made by the developer instead of the user, as such, I believe the server should halt violently with tracebacks instead of continue running.